src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -361,10 +361,7 @@ def is_already_paid(self, order, installment):
                     self._do_on_payment_success(order, payment)
                     return True
 
-                if status == "UNPAID":
-                    self._do_on_payment_failure(order, installment["id"])
-                    return False
-
+        order.set_installment_refused(installment["id"])
```

Comment:
I wonder if we should not continue to set the installment to `refused` state only if an `UNPAID` transaction is found. But we should continue to iterate over the transactions list.
Actually, if a transaction has never been debited, its state will be set to `refused` first.